react-native-mcu-manager/android/src/main/java/uk/co/playerdata/reactnativemcumanager/ZipPackage.java

```
@@ -0,0 +1,179 @@
+             */
+            private int imageIndex = 0;
+            /**
+             * The slot number where the image is to be sent. By default images are sent to the
+             * secondary slot and then swapped to the primary slot after the image is confirmed
+             * and the device is reset.
+             * <p>
+             * However, if the device supports Direct XIP feature it is possible to run an app
+             * from a secondary slot. The image has to be compiled for this slot. A ZIP package
+             * can contain images for both slots. Only the one targeting the available one will
+             * be sent.
+             * @since NCS v 2.5, nRF Connect Device Manager 1.8.
+             */
+            private int slot = TargetImage.SLOT_SECONDARY;
+        }
+    }
+
+    private Manifest manifest;
+    private final Map<String, byte[]> entries = new HashMap<>();
+
+    public ZipPackage(@NonNull final byte[] data) throws IOException {
+        ZipEntry ze;
+
+        // Unzip the file and look for the manifest.json.
+        final ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(data));
+        while ((ze = zis.getNextEntry()) != null) {
+            if (ze.isDirectory())
+                throw new IOException("Invalid ZIP");
+
+            final String name = validateFilename(ze.getName(), ".");
+
+            if (name.equals(MANIFEST)) {
+                final Gson gson = new GsonBuilder()
+                        .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+                        .create();
+                manifest = gson.fromJson(new InputStreamReader(zis), Manifest.class);
+            } else if (name.endsWith(".bin") || name.endsWith(".suit")) {
+                final byte[] content = getData(zis);
+                entries.put(name, content);
+            } else {
+                Log.v("KT", "les");
```

what is this log trying to communicate?

---

@saty9 It's a mistake, they use Timber log on [the original file ](https://github.com/NordicSemiconductor/Android-nRF-Connect-Device-Manager/blob/dee599274d2e08d83a6231bf434a35ab4b4414cd/sample/src/main/java/io/runtime/mcumgr/sample/utils/ZipPackage.java#L91), I can remove it.

Or what do you think about throwing a new error instead of logs?
`throw new IOException("Unsupported file found: %s " + name);`

---

I think throwing seems reasonable although is %s and + the way to insert a string in java?

---

Fix with `throw new IOException("Unsupported file found: " + name);`